frontends/concrete-python/concrete/fhe/compilation/compiler.py

```diff
@@ -456,6 +457,20 @@ def compile(
             )
         )
 
+        # if using tfhers integers, parameter selection strategy has to be multi-parameters.
+        # we try to catch this early, but the compiler will also fail and provide some info about
+        # it. So we don't need to scan all the inputset here.
+        if (
+            check_inputset_has_element_of_type(inputset, TFHERSInteger, search_all=False)
```

Comment:
Not sure that is true ? As example we could have on TFHERSInteger in a circuit and should works... But not sure we want to support those kind of programs

Reply:
it's not that the computation isn't correct, but because we will be adding a change_partition operation, we are already expecting to be in the multi-parameter setting, so it's safer fail here with a clear error message, than to leave the compiler complain about a change_partition operation in a mono-parameter setting.